pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -198,10 +199,18 @@ void closeAllConnections() {
                 // succeed
                 future.thenAccept(ClientCnx::close);
             }
-        }));
+        });
     }
 
-    /**
+    private String getKey(InetSocketAddress logicalAddress,
+                          InetSocketAddress physicalAddress, final int randomKey) {
+        StringJoiner sj = new StringJoiner("#");
+        sj.add(logicalAddress.toString());
+        sj.add(physicalAddress.toString());
+        sj.add(String.valueOf(randomKey));
+        return sj.toString();
+    }
```

Comment:
A better approach is to use a `record` or an immutable class with the fields and with a `hashCode` and `equals` implementation (Lombok's `@Value` handles this too, but `record` is preferred.)

Reply:
Sure. Will do that.

Reply:
Updated it to use Lombok.Value in case we need to backport this to earlier versions(<3.0).